release-notes-1.2.0.md

```diff
@@ -339,6 +339,17 @@ The above-mentioned lang library modules have been enhanced by adding new API el
 - `lang.array` module - New functions added: `lastIndexOf()`, `toStream()`
 - `lang.map` module - New functions added: `toArray()`, `removeIfHasKey()`
 
+
+### Backward incompatible improvements and bug fixes
+- Listener variables are now final.
+- The syntax for Unicode escapes in strings has changed from `\u[CodePoint]` to `\u{CodePoint}` so as to align with ECMAScript. Although this is an incompatible change, the previous syntax was not implemented.
+- The semantics of the `lock` statement has been changed. Now it acquires a single, program-wide, recursive mutex before executing a lock statement, and releases the mutex after completing the execution of the lock statement.
+- When a list constructor or a mapping constructor is used without a contextually expected type, we now infer a tuple or a record type rather than an array or a map type.
```

Comment:
```suggestion
- When a list constructor or a mapping constructor is used without a contextually-expected type, now a tuple or a record type is inferred rather than an array or a map type.
```